lib/commands/connect/info.js

```diff
@@ -23,18 +14,22 @@ module.exports = {
     ],
     needsApp: true,
     needsAuth: true,
-    run: cli.command(function (context, heroku) {
-      return api.withUserConnections(context.auth.password, context.app, context.flags, true, heroku).then(function(connections){
-          if (connections.length === 0) {
-            console.log("No connection found, requesting auth...");
-            api.requestAppAccess(context.auth.password, context.app).then(function() {
-              api.withUserConnections(context.auth.password, context.app, context.flags, false, heroku).then(function(connections){
-                connections_info(connections);
-              });          
-            });
-          } else {
-            connections_info(connections);
-          }
+    run: cli.command(co.wrap(function* (context, heroku) {
+      let connections = yield api.withUserConnections(context.auth.password, context.app, context.flags, true, heroku);
```

Comment:
I'd probably have this be a var, since the scope is the function and then it's more clear what is happening.